src/libmeasurement_kit/ooni/telegram.cpp

```diff
@@ -0,0 +1,194 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include "../common/utils.hpp"
+#include "../ooni/constants.hpp"
+#include <measurement_kit/ooni.hpp>
+
+namespace mk {
+namespace ooni {
+
+using namespace mk::report;
+
+static void tcp_many(const std::vector<std::string> ip_ports,
+                     Var<Entry> entry,
+                     Callback<Error> cb,
+                     Var<Reactor> reactor,
+                     Var<Logger> logger) {
+    // C++20 should have continuations in std; until then,
+    // we copy the ghetto pattern from web_connectivity
```

Comment:
BTW, we also have an abstraction called `Continuation` that basically is a suspended function that can be resumed at a later point in time. See its [an example of usage](https://github.com/measurement-kit/measurement-kit/blob/master/src/libmeasurement_kit/report/file_reporter.cpp#L40); perhaps it can also be useful in this case?